### ISSUE TITLE
🐞 `TypeError: closeButton is undefined` when Esc button is pressed when Debug Bar plugin is installed

### ISSUE THREAD

**Issue by stefanos82** [ClassicPress/ClassicPress#1487]
### Expected behavior

It shouldn't complain when Debug Bar info window is NOT opened, else if it happens to be open, then it should close it.

### Current behavior

![image](https://github.com/ClassicPress/ClassicPress/assets/1777466/25f4666b-dcdb-40e6-9df1-05a8c84d8480)

### Steps to reproduce bug

1. Install the aforementioned plugin
2. Press Esc button
3. Click on the Debug menu located in the upper right corner that just became red

### Context

About PHP version, as I cannot choose my current installed version from below: I use `PHP 8.2.21`

### ClassicPress version

2.1.1+dev (WP-6.2.6) [f765ecadfe]

### PHP version

I don't know

### Can you help?

No, I can't help

**Comment by mattyrob** [comment]
Confirmed bug.

In `wp-admin/js/plugin-install.js` there is code that adds a listener to remove the iframe by pressing the escape key. I'm not sure that this is needed as the same listener code is added further up.

If it needs to remain the the presence of `closeButton` needs to be checked first. Maybe something lik:
```
closeButton = dialog.querySelector( '#dialog-close-button' );
if ( closeButton !== null ) {
	closeButton.remove();
}
```

**Comment by KTS915** [comment]
I can't check the solution right now but, in the above code, it would be better to have
```
if ( closeButton != null ) {
```
as that checks for `undefined` as well as `null`.

**Comment by stefanos82** [comment]
It's better to be more strict, because JavaScript is...well, JavaScript! :rofl: 

```js
if ( closeButton !== null || closeButton !== undefined ) {
```

**Comment by KTS915** [comment]
>It's better to be more strict

Not really. What I wrote is the generally accepted way of testing for both conditions in JavaScript.

**Comment by ginsterbusch** [comment]
Just go for `typeof( closeButton ) != 'undefined' && !!closeButton`

**Comment by xxsimoxx** [comment]
> It shouldn't complain when Debug Bar info window is NOT opened

This is an issue and the above proposed change fix it

> else if it happens to be open, then it should close it.

I've tried on a WP installation and also there pressing esc don't work, so if confirmed it's a plugin's bug.

**Comment by xxsimoxx** [comment]
Re-tested on WP 6.5.5 and WP 6.6, Debug Bar 1.1.6, Chrome 126.0.6478.127 (Build ufficiale) (arm64).
Pressing esc isn't working here.

**Comment by stefanos82** [comment]
Weird...anyway, seems like it's unrelated to ClassicPress then; closing.

**Comment by mattyrob** [comment]
I can confirm this issue in Debug bar but also in the browser console.

ClassicPress has refactored many JavaScript files so behaviour may not be equivalent to WordPress. I believe that this is a ClassicPress bug that needs a fix as proposed in #1493

**Comment by mattyrob** [comment]
Fixed with merge of #1493
